src/lib/pickles/impls.ml

```diff
@@ -36,11 +40,15 @@ module Step = struct
       Field.t * Boolean.var
 
     let forbidden_shifted_values =
-      forbidden_shifted_values ~size_in_bits:Constant.size_in_bits
-        ~modulus:(Wrap_impl.Bigint.to_bignum_bigint Constant.size) ~f:(fun x ->
-          let hi = test_bit x (Field.size_in_bits - 1) in
-          let lo = B.shift_right x 1 in
-          (Impl.Bigint.(to_field (of_bignum_bigint lo)), hi))
+      let f x =
+        let hi = test_bit x (Field.size_in_bits - 1) in
+        let lo = B.shift_right x 1 in
+        (Impl.Bigint.(to_field (of_bignum_bigint lo)), hi)
+      in
+      let modulus = Wrap_impl.Bigint.to_bignum_bigint Constant.size in
+      let size_in_bits = Constant.size_in_bits in
+      let values = forbidden_shifted_values ~size_in_bits ~modulus in
+      values |> List.filter ~f:(fun x -> B.compare modulus x > 0) |> List.map ~f
```

Comment:
Instead of `filter` here, can you `map` and cast any too-large values to be `0` instead? That should preserve the old behaviour.